reactor-netty-http/src/main/java/reactor/netty/http/server/HttpCompressionSettingsSpec.java

```diff
@@ -0,0 +1,178 @@
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package reactor.netty.http.server;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.netty.handler.codec.compression.Brotli;
+import io.netty.handler.codec.compression.BrotliOptions;
+import io.netty.handler.codec.compression.CompressionOptions;
+import io.netty.handler.codec.compression.DeflateOptions;
+import io.netty.handler.codec.compression.GzipOptions;
+import io.netty.handler.codec.compression.SnappyOptions;
+import io.netty.handler.codec.compression.StandardCompressionOptions;
+import io.netty.handler.codec.compression.Zstd;
+import io.netty.handler.codec.compression.ZstdOptions;
+import io.netty.util.internal.ObjectUtil;
+
+/**
+ * HTTP Compression configuration builder for the {@link SimpleCompressionHandler}.
+ *
+ * @author raccoonback
+ */
+public final class HttpCompressionSettingsSpec {
+
+	private final GzipOptions gzipOptions;
+	private final DeflateOptions deflateOptions;
+	private final SnappyOptions snappyOptions;
+	private BrotliOptions brotliOptions;
+	private ZstdOptions zstdOptions;
+
+	private HttpCompressionSettingsSpec() {
+		gzipOptions = StandardCompressionOptions.gzip();
+		deflateOptions = StandardCompressionOptions.deflate();
+		snappyOptions = StandardCompressionOptions.snappy();
+
+		if (Brotli.isAvailable()) {
+			brotliOptions = StandardCompressionOptions.brotli();
+		}
+
+		if (Zstd.isAvailable()) {
+			zstdOptions = StandardCompressionOptions.zstd();
+		}
+	}
+
+	private HttpCompressionSettingsSpec(Build build) {
+		gzipOptions = build.gzipOptions;
+		deflateOptions = build.gzipOptions;
+		snappyOptions = StandardCompressionOptions.snappy();
+
+		if (Brotli.isAvailable()) {
+			brotliOptions = StandardCompressionOptions.brotli();
+		}
+
+		if (Zstd.isAvailable() && build.zstdOptions != null) {
+			zstdOptions = build.zstdOptions;
+		}
+	}
+
+	/**
+	 * Creates a builder for {@link HttpCompressionSettingsSpec}.
+	 *
+	 * @return a new {@link HttpCompressionSettingsSpec.Builder}
+	 */
+	public static Builder builder() {
+		return new Build();
+	}
+
+	static HttpCompressionSettingsSpec provideDefault() {
+		return new HttpCompressionSettingsSpec();
+	}
+
+	CompressionOptions[] adaptToOptions() {
+		List<CompressionOptions> options = new ArrayList<>();
+		options.add(this.gzipOptions);
+		options.add(this.deflateOptions);
+		options.add(this.snappyOptions);
+
+		if (brotliOptions != null) {
+			options.add(this.brotliOptions);
+		}
+
+		if (zstdOptions != null) {
+			options.add(this.zstdOptions);
+		}
+
+		return options.toArray(new CompressionOptions[0]);
+	}
```

Comment:
Adapts to the type to be provided to the constructor argument of Netty's [`HttpContentCompressor`](https://github.com/netty/netty/blob/bbf3efec982d9eb22cc8ca433b17ec052cb76d59/codec-http/src/main/java/io/netty/handler/codec/http/HttpContentCompressor.java#L160-L162).